src/main/java/com/yanna/stepanova/service/impl/UserServiceImpl.java

```diff
@@ -4,23 +4,30 @@
 import com.yanna.stepanova.dto.user.UserResponseDto;
 import com.yanna.stepanova.exception.RegistrationException;
 import com.yanna.stepanova.mapper.UserMapper;
+import com.yanna.stepanova.model.User;
+import com.yanna.stepanova.repository.user.RoleRepository;
 import com.yanna.stepanova.repository.user.UserRepository;
 import com.yanna.stepanova.service.UserService;
 import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @RequiredArgsConstructor
 @Service
 public class UserServiceImpl implements UserService {
     private final UserRepository userRepo;
     private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
+    private final RoleRepository roleRepo;
 
     @Override
     public UserResponseDto register(UserRegistrationRequestDto requestDto)
             throws RegistrationException {
         if (userRepo.existsByEmail(requestDto.email())) {
             throw new RegistrationException("This user can't be registered");
         }
-        return userMapper.toResponseDto(userRepo.save(userMapper.toModel(requestDto)));
+        User user = userMapper.toModel(requestDto);
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
```

Comment:
Do we need to set up Role.User during the registration or you are doing this in other way?